api/src/main/java/org/openmrs/api/impl/DiagnosisServiceImpl.java

```diff
@@ -174,4 +174,19 @@ public DiagnosisDAO getDiagnosisDAO() {
 	public void setDiagnosisDAO(DiagnosisDAO diagnosisDAO) {
 		this.diagnosisDAO = diagnosisDAO;
 	}
+	
+
+    /**
+     * Finds the diagnoses for a given encounter
+     * 
+     * @param encounter the encounter whose diagnoses we are to get
+     * 
+     * @return the list of diagnoses in the given encounter 
+    */
+
+	@Override
+	public List<Diagnosis> getDiagnoses(Encounter encounter) {
```

Comment:
Thanks @HerbertYiga  for working on this, it seems this method was implemented in [hibernateDiagnosisDao](https://github.com/openmrs/openmrs-core/blob/a789b485dd0d11bfc93b32ca21a5a91bf787b8a0/api/src/main/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAO.java#L95-L100),however am not sure of what you are trying to achieve but basically this method is supposed to return list of all diagnosis drugs in an encounter

Reply:
@sherrif10  thanks Sharif, for this case i am only trying to expose the already existing method get Diagnoses(Encounter encounter)  but not re-defining its implementation as seen at the line having this (return diagnosisDAO.getDiagnoses(encounter);)